.github/workflows/build.yml

```diff
@@ -53,30 +52,87 @@ jobs:
       - name: Upload test coverage
         uses: codecov/codecov-action@v1
 
-      - name: Is Release?
-        if: startswith(github.ref, 'refs/tags/v')
-        run: echo "DEPLOY_PACKAGE=true" >> $GITHUB_ENV
+  publish:
+    needs: build
+    if: startswith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-latest
+    environment: production
+    env:
+      NODE_VER: 16.14.0
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Node ${{ env.NODE_VER }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ env.NODE_VER }}
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Build package
+        run: npm run build-ci
 
-      # note: package.json gets updated here for the new package name
       - name: Publish to npm (@dapr/dapr)
-        if: env.DEPLOY_PACKAGE == 'true'
         run: npm publish build/ --access public
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
+      # dapr-client is deprecated, but we still want to publish it for a while.
       - name: "[dapr-client] Configure to publish to dapr-client for deprecation notice reasons"
-        if: env.DEPLOY_PACKAGE == 'true'
         run: |
           sed -i s#"@dapr/dapr"#"dapr-client"# package.json
           echo "This has been deprecated and will not be updated anymore, please use https://www.npmjs.com/package/@dapr/dapr" > README.md
 
+      - name: "[dapr-client] Install dependencies"
+        run: npm install
+
       - name: "[dapr-client] Build Package"
-        if: env.DEPLOY_PACKAGE == 'true'
-        run: npm run build
+        run: npm run build-ci
 
-      # note: package.json gets updated here for the new package name
       - name: "[dapr-client] Publish to npm (dapr-client)"
-        if: env.DEPLOY_PACKAGE == 'true'
+        run: npm publish build/ --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  publish-dev:
+    needs: build
```

Comment:
Please help me understand: `publish-dev` needs `build` job to be done, and in `build` job, we have `Build package (install dependencies, lint, prettier, build)` - so, are we checking lint and prettier for publish-dev as well?

Reply:
Added the `needs` dependency to run it sequentially, so that if build (lint/pretty) or unit tests fail, we don't even start publishing.